cmake/std/atdm/test/unit_tests/set_build_options_unit_tests.sh

```diff
@@ -0,0 +1,484 @@
+
+  ATDM_CONFIG_BUILD_NAME=default_no-rdc
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_CUDA_RDC} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default_rdc_no-rdc
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_CUDA_RDC} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default-rdc-no-rdc
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_CUDA_RDC} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default_rdc_no-rdc-after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_CUDA_RDC} OFF
+
+}
+
+
+testFPIC() {
+
+  ATDM_CONFIG_BUILD_NAME=default
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default-fpic-after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} ON
+
+  ATDM_CONFIG_BUILD_NAME=default_fpic-after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_FPIC} ON
+
+}
+
+
+testComplex() {
+
+  ATDM_CONFIG_BUILD_NAME=default
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default-complex
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} ON
+
+  ATDM_CONFIG_BUILD_NAME=default-COMPLEX
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} ON
+
+  ATDM_CONFIG_BUILD_NAME=default-complexs
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default_complex_after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} ON
+
+  ATDM_CONFIG_BUILD_NAME=default-no-complex
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default-nos-complex
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} ON
+
+  ATDM_CONFIG_BUILD_NAME=default-no-complex-after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default-no-COMPLEX-after
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_COMPLEX} OFF
+
+}
+
+
+testSharedLibs() {
+
+  ATDM_CONFIG_BUILD_NAME=default
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_SHARED_LIBS} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default-static
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_SHARED_LIBS} OFF
+
+  ATDM_CONFIG_BUILD_NAME=default-static
+  . ${ATDM_CONFIG_SCRIPT_DIR}/utils/set_build_options.sh
+  ${_ASSERT_EQUALS_} ${ATDM_CONFIG_SHARED_LIBS} OFF
```

Comment:
Is this a duplicate of the one above?

Reply:
Shoot, you are right, it is duplicated.  I will remove the extra.